sa/storage-authority.go

```diff
@@ -636,3 +636,33 @@ func (ssa *SQLStorageAuthority) AlreadyDeniedCSR(names []string) (already bool,
 
 	return
 }
+
+// GetSCTReceipt gets a specific SCT receipt for a given certificate serial and
+// CT log ID
+func (ssa *SQLStorageAuthority) GetSCTReceipt(serial string, logID string) (receipt core.SignedCertificateTimestamp, err error) {
+	err = ssa.dbMap.SelectOne(
+		&receipt,
+		"SELECT * FROM sctReceipts WHERE certificateSerial = :serial AND logID = :logID",
+		map[string]interface{}{
+			"serial": serial,
+			"logID":  logID,
+		},
+	)
+
+	return
+}
+
+// AddSCTReceipt adds a new SCT receipt to the (append-only) sctReceipts table
+func (ssa *SQLStorageAuthority) AddSCTReceipt(sct core.SignedCertificateTimestamp) error {
+	tx, err := ssa.dbMap.Begin()
```

Comment:
I don't think you need a transaction here, since you are making only the one change, and it doesn't depend on data read previously from the DB, nor does it depend on other changes being committed at the same time.